eclipse/kura
# feat(api): added gnss type data to PositionService (#5381) [backport release-5.6.0]

Backport of #5381 

--------------------------------------------------------------------

* feat(api): added gnss type data to PositionService

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>

* test: added tests

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>

* refactor: moved GNSSType enum to api bundle

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>

* refactor: removed testing logger

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>

* refactor: refactored gnsstype

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>

* refactor: updated with suggestions

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>

* fix: fixed line length

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>

* Update kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/position/PositionService.java

Co-authored-by: Mattia Dal Ben <mattdibi@users.noreply.github.com>

* Update GNSSType.java

---------

Signed-off-by: SimoneFiorani <simone.fiorani@abinsula.com>
Co-authored-by: Mattia Dal Ben <mattdibi@users.noreply.github.com>
(cherry picked from commit 51fcd57437af67f22809b24ccc116b52c347c336)

> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped

**Screenshots:** If applicable, add screenshots to help explain your solution

**Manual Tests**: Optional description of the tests performed to check correct functioning of changes, useful for an efficient review

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]
